bug: WebViewパネル作成時にactiveTextEditorがundefinedになりタスク取得に失敗する

## 現象

WebViewパネルを開いた状態で、アクティブなMarkdownエディタがない場合に以下のエラーログが出力される：

```
[ERROR] Error: Failed to parse task at line 0: アクティブなエディタがありません
{errorType: 'TaskParseError', message: 'Failed to parse task at line 0: アクティブなエディタがありません'}
```

## 原因分析

### エラーの発生フロー

1. WebViewが`GET_TASKS`メッセージを送信（初期化時やエディタ切り替え時）
2. `webViewMessageHandler.handleGetTasks()` → `taskController.getTasks()`
3. `markdownTaskRepository.findAll()` が実行される
4. `vscodeDocumentClient.getActiveDocumentText()` がアクティブエディタがない場合に `DocumentNotFoundError` を返す
5. `markdownTaskRepository.ts:26` で `TaskParseError(0, textResult.error.message)` に変換される
6. `webViewMessageHandler.ts:145` の `sendError()` で `logger.error()` が呼ばれる

### 問題点

1. **エラー型が不適切**: 「アクティブなエディタがない」は「タスクのパースに失敗した」とは異なる状況。`TaskParseError` ではなく専用のエラー型（例: `NoActiveEditorError`）が適切

2. **エラーログレベルが不適切**: アクティブエディタがないのは正常な状態遷移の一つ（ユーザーが別のタブを見ている、設定画面を開いている等）。`logger.error` ではなく `logger.debug` または `logger.info` が適切

3. **ユーザー体験**: WebViewに「アクティブなエディタがありません」というエラーメッセージを表示するのではなく、「Markdownファイルを開いてください」のような案内が適切

### 発生シナリオ

- VSCode起動直後（エディタがまだ開いていない）
- WebViewパネルを開いた状態で、Markdownファイルのエディタを閉じた
- 別のタブ（設定画面、拡張機能画面など非テキストエディタ）に切り替えた

## 関連ファイル

- `packages/core/src/infrastructure/clients/vscodeDocumentClient.ts:66` - エラー発生元
- `packages/core/src/infrastructure/adapters/markdownTaskRepository.ts:26` - TaskParseErrorへの変換
- `packages/core/src/interface/adapters/webViewMessageHandler.ts:145` - エラーログ出力

## 提案する修正

1. `NoActiveEditorError` または `EditorNotReadyError` のような専用エラー型を追加
2. `markdownTaskRepository` でエラー型を適切に伝播
3. `webViewMessageHandler` でエラー型に応じてログレベルを調整
4. WebView側で適切なメッセージ（「Markdownファイルを開いてください」）を表示

--- a/packages/core/src/application/usecases/changeTaskStatusUseCase.ts
+++ b/packages/core/src/application/usecases/changeTaskStatusUseCase.ts
@@ -1,5 +1,6 @@
 import type { Result } from 'neverthrow';
 import type { Task } from '../../domain/entities/task';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import type { TaskParseError } from '../../domain/errors/taskParseError';
 import type { ConfigProvider } from '../../domain/ports/configProvider';
@@ -21,7 +22,7 @@ export class ChangeTaskStatusUseCase {
 	async execute(
 		id: string,
 		newStatus: Status,
-	): Promise<Result<Task, TaskNotFoundError | TaskParseError>> {
+	): Promise<Result<Task, TaskNotFoundError | TaskParseError | NoActiveEditorError>> {
 		// 設定を取得
 		const config = await this.configProvider.getConfig();
 

--- a/packages/core/src/application/usecases/createTaskUseCase.ts
+++ b/packages/core/src/application/usecases/createTaskUseCase.ts
@@ -1,5 +1,6 @@
 import type { Result } from 'neverthrow';
 import { Task, type TaskMetadata } from '../../domain/entities/task';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import type { ConfigProvider } from '../../domain/ports/configProvider';
 import type { TaskRepository } from '../../domain/ports/taskRepository';
@@ -28,7 +29,9 @@ export class CreateTaskUseCase {
 	/**
 	 * 新しいタスクを作成する
 	 */
-	async execute(input: CreateTaskInput): Promise<Result<Task, TaskNotFoundError>> {
+	async execute(
+		input: CreateTaskInput,
+	): Promise<Result<Task, TaskNotFoundError | NoActiveEditorError>> {
 		const config = await this.configProvider.getConfig();
 
 		// ステータスが指定されていない場合はデフォルトステータスを使用

--- a/packages/core/src/application/usecases/deleteTaskUseCase.ts
+++ b/packages/core/src/application/usecases/deleteTaskUseCase.ts
@@ -1,4 +1,5 @@
 import type { Result } from 'neverthrow';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import type { TaskRepository } from '../../domain/ports/taskRepository';
 
@@ -11,7 +12,7 @@ export class DeleteTaskUseCase {
 	/**
 	 * タスクを削除する
 	 */
-	async execute(id: string): Promise<Result<void, TaskNotFoundError>> {
+	async execute(id: string): Promise<Result<void, TaskNotFoundError | NoActiveEditorError>> {
 		return this.taskRepository.delete(id);
 	}
 }

--- a/packages/core/src/application/usecases/getTasksUseCase.ts
+++ b/packages/core/src/application/usecases/getTasksUseCase.ts
@@ -1,5 +1,6 @@
 import type { Result } from 'neverthrow';
 import type { Task } from '../../domain/entities/task';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskParseError } from '../../domain/errors/taskParseError';
 import type { TaskRepository } from '../../domain/ports/taskRepository';
 import type { Path } from '../../domain/valueObjects/path';
@@ -13,14 +14,14 @@ export class GetTasksUseCase {
 	/**
 	 * 全タスクを取得する
 	 */
-	async execute(): Promise<Result<Task[], TaskParseError>> {
+	async execute(): Promise<Result<Task[], TaskParseError | NoActiveEditorError>> {
 		return this.taskRepository.findAll();
 	}
 
 	/**
 	 * 指定したパス配下のタスクを取得する
 	 */
-	async executeByPath(path: Path): Promise<Result<Task[], TaskParseError>> {
+	async executeByPath(path: Path): Promise<Result<Task[], TaskParseError | NoActiveEditorError>> {
 		return this.taskRepository.findByPath(path);
 	}
 }

--- a/packages/core/src/application/usecases/updateTaskUseCase.ts
+++ b/packages/core/src/application/usecases/updateTaskUseCase.ts
@@ -1,5 +1,6 @@
 import type { Result } from 'neverthrow';
 import type { Task, TaskMetadata } from '../../domain/entities/task';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import type { TaskParseError } from '../../domain/errors/taskParseError';
 import type { TaskRepository } from '../../domain/ports/taskRepository';
@@ -24,7 +25,9 @@ export class UpdateTaskUseCase {
 	/**
 	 * タスクを更新する
 	 */
-	async execute(input: UpdateTaskInput): Promise<Result<Task, TaskNotFoundError | TaskParseError>> {
+	async execute(
+		input: UpdateTaskInput,
+	): Promise<Result<Task, TaskNotFoundError | TaskParseError | NoActiveEditorError>> {
 		// タスクを取得
 		const findResult = await this.taskRepository.findById(input.id);
 		if (findResult.isErr()) {

--- a/packages/core/src/bootstrap/di/container.ts
+++ b/packages/core/src/bootstrap/di/container.ts
@@ -175,6 +175,13 @@ export class Container {
 	getConfigController(): ConfigController {
 		return this.configController;
 	}
+
+	/**
+	 * VscodeDocumentClientを取得
+	 */
+	getVscodeDocumentClient(): VscodeDocumentClient {
+		return this.vscodeDocumentClient;
+	}
 }
 
 /**

--- a/packages/core/src/bootstrap/webview/kanbanPanelProvider.ts
+++ b/packages/core/src/bootstrap/webview/kanbanPanelProvider.ts
@@ -21,6 +21,17 @@ export class KanbanPanelProvider {
 	}
 
 	/**
+	 * 現在のアクティブエディタがMarkdownファイルの場合、そのURIを保存する
+	 */
+	private updateCurrentDocumentUri(): void {
+		const editor = vscode.window.activeTextEditor;
+		if (editor && this.isMarkdownDocument(editor.document)) {
+			this.container.getVscodeDocumentClient().setCurrentDocumentUri(editor.document.uri);
+			logger.debug(`Current document URI updated: ${editor.document.uri.toString()}`);
+		}
+	}
+
+	/**
 	 * パネルを表示または作成する
 	 */
 	public showOrCreate(viewColumn?: vscode.ViewColumn): void {
@@ -29,6 +40,10 @@ export class KanbanPanelProvider {
 			this.panel.reveal(viewColumn);
 			return;
 		}
+
+		// パネル作成前に現在のドキュメントURIを保存
+		// (createWebviewPanelの後はactiveTextEditorがundefinedになるため)
+		this.updateCurrentDocumentUri();
 
 		// 新しいパネルを作成
 		this.panel = vscode.window.createWebviewPanel(
@@ -126,6 +141,9 @@ export class KanbanPanelProvider {
 		// アクティブエディタ変更時
 		const activeEditorChange = vscode.window.onDidChangeActiveTextEditor(async (editor) => {
 			if (editor && this.isMarkdownDocument(editor.document)) {
+				// MarkdownファイルのURIを更新
+				this.container.getVscodeDocumentClient().setCurrentDocumentUri(editor.document.uri);
+				logger.debug(`Current document URI updated: ${editor.document.uri.toString()}`);
 				await this.sendTasksUpdate();
 			}
 		});
@@ -133,11 +151,17 @@ export class KanbanPanelProvider {
 
 		// ドキュメント変更時
 		const documentChange = vscode.workspace.onDidChangeTextDocument(async (event) => {
+			// currentDocumentUriがセットされている場合は、そのドキュメントの変更を検出
+			const currentUri = this.container.getVscodeDocumentClient().getCurrentDocumentUri();
+			const isCurrentDocument =
+				currentUri && event.document.uri.toString() === currentUri.toString();
+
+			// アクティブエディタのドキュメントまたはcurrentDocumentの変更を検出
 			const activeEditor = vscode.window.activeTextEditor;
-			if (activeEditor && event.document === activeEditor.document) {
-				if (this.isMarkdownDocument(event.document)) {
-					await this.sendTasksUpdate();
-				}
+			const isActiveDocument = activeEditor && event.document === activeEditor.document;
+
+			if ((isCurrentDocument || isActiveDocument) && this.isMarkdownDocument(event.document)) {
+				await this.sendTasksUpdate();
 			}
 		});
 		this.disposables.push(documentChange);

--- a/packages/core/src/domain/errors/index.ts
+++ b/packages/core/src/domain/errors/index.ts
@@ -1,4 +1,5 @@
 // Domain Errors
 export { InvalidStatusError } from './invalidStatusError';
+export { NoActiveEditorError } from './noActiveEditorError';
 export { TaskNotFoundError } from './taskNotFoundError';
 export { TaskParseError } from './taskParseError';

--- a/packages/core/src/domain/errors/noActiveEditorError.ts
+++ b/packages/core/src/domain/errors/noActiveEditorError.ts
@@ -1,0 +1,12 @@
+/**
+ * アクティブなエディタがないエラー
+ * これは正常な状態遷移の一つ（WebViewにフォーカスがある場合など）
+ */
+export class NoActiveEditorError extends Error {
+	readonly _tag = 'NoActiveEditorError';
+
+	constructor(message = 'Markdownファイルを開いてください') {
+		super(message);
+		this.name = 'NoActiveEditorError';
+	}
+}

--- a/packages/core/src/domain/ports/taskRepository.ts
+++ b/packages/core/src/domain/ports/taskRepository.ts
@@ -1,5 +1,6 @@
 import type { Result } from 'neverthrow';
 import type { Task } from '../entities/task';
+import type { NoActiveEditorError } from '../errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../errors/taskNotFoundError';
 import type { TaskParseError } from '../errors/taskParseError';
 import type { Path } from '../valueObjects/path';
@@ -12,30 +13,32 @@ export interface TaskRepository {
 	/**
 	 * 全タスクを取得する
 	 */
-	findAll(): Promise<Result<Task[], TaskParseError>>;
+	findAll(): Promise<Result<Task[], TaskParseError | NoActiveEditorError>>;
 
 	/**
 	 * IDでタスクを取得する
 	 */
-	findById(id: string): Promise<Result<Task, TaskNotFoundError | TaskParseError>>;
+	findById(
+		id: string,
+	): Promise<Result<Task, TaskNotFoundError | TaskParseError | NoActiveEditorError>>;
 
 	/**
 	 * パスでタスクをフィルタリングして取得する
 	 */
-	findByPath(path: Path): Promise<Result<Task[], TaskParseError>>;
+	findByPath(path: Path): Promise<Result<Task[], TaskParseError | NoActiveEditorError>>;
 
 	/**
 	 * タスクを保存する（作成または更新）
 	 */
-	save(task: Task): Promise<Result<Task, TaskNotFoundError>>;
+	save(task: Task): Promise<Result<Task, TaskNotFoundError | NoActiveEditorError>>;
 
 	/**
 	 * タスクを削除する
 	 */
-	delete(id: string): Promise<Result<void, TaskNotFoundError>>;
+	delete(id: string): Promise<Result<void, TaskNotFoundError | NoActiveEditorError>>;
 
 	/**
 	 * 利用可能なパス（見出し階層）を全て取得する
 	 */
-	getAvailablePaths(): Promise<Result<Path[], TaskParseError>>;
+	getAvailablePaths(): Promise<Result<Path[], TaskParseError | NoActiveEditorError>>;
 }

--- a/packages/core/src/infrastructure/adapters/markdownTaskRepository.test.ts
+++ b/packages/core/src/infrastructure/adapters/markdownTaskRepository.test.ts
@@ -38,6 +38,7 @@ const createMockVscodeDocumentClient = (
 ): VscodeDocumentClient =>
 	({
 		getActiveDocumentText: vi.fn().mockReturnValue(ok('# Test')),
+		getCurrentDocumentText: vi.fn().mockResolvedValue(ok('# Test')),
 		replaceDocumentText: vi.fn().mockResolvedValue(ok(undefined)),
 		...overrides,
 	}) as unknown as VscodeDocumentClient;
@@ -104,7 +105,7 @@ describe('MarkdownTaskRepository', () => {
 		it('ドキュメントがない場合はエラーを返す', async () => {
 			const markdownClient = createMockMarkdownTaskClient();
 			const documentClient = createMockVscodeDocumentClient({
-				getActiveDocumentText: vi.fn().mockReturnValue({
+				getCurrentDocumentText: vi.fn().mockResolvedValue({
 					isErr: () => true,
 					isOk: () => false,
 					error: { message: 'No document' },

--- a/packages/core/src/infrastructure/adapters/markdownTaskRepository.ts
+++ b/packages/core/src/infrastructure/adapters/markdownTaskRepository.ts
@@ -1,11 +1,13 @@
 import { err, ok, type Result } from 'neverthrow';
 import { Task } from '../../domain/entities/task';
+import { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import { TaskParseError } from '../../domain/errors/taskParseError';
 import type { TaskRepository } from '../../domain/ports/taskRepository';
 import type { Path } from '../../domain/valueObjects/path';
 import type { MarkdownTaskClient, ParsedTask } from '../clients/markdownTaskClient';
 import type { VscodeDocumentClient } from '../clients/vscodeDocumentClient';
+import { NoActiveEditorError as InfraNoActiveEditorError } from '../clients/vscodeDocumentClient';
 
 /**
  * MarkdownTaskRepository
@@ -20,9 +22,13 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * 全タスクを取得する
 	 */
-	async findAll(): Promise<Result<Task[], TaskParseError>> {
-		const textResult = this.documentClient.getActiveDocumentText();
+	async findAll(): Promise<Result<Task[], TaskParseError | NoActiveEditorError>> {
+		const textResult = await this.documentClient.getCurrentDocumentText();
 		if (textResult.isErr()) {
+			// NoActiveEditorErrorの場合はドメイン層のエラーに変換
+			if (textResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskParseError(0, textResult.error.message));
 		}
 
@@ -38,7 +44,9 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * IDでタスクを取得する
 	 */
-	async findById(id: string): Promise<Result<Task, TaskNotFoundError | TaskParseError>> {
+	async findById(
+		id: string,
+	): Promise<Result<Task, TaskNotFoundError | TaskParseError | NoActiveEditorError>> {
 		const allResult = await this.findAll();
 		if (allResult.isErr()) {
 			return err(allResult.error);
@@ -55,7 +63,7 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * パスでタスクをフィルタリングして取得する
 	 */
-	async findByPath(path: Path): Promise<Result<Task[], TaskParseError>> {
+	async findByPath(path: Path): Promise<Result<Task[], TaskParseError | NoActiveEditorError>> {
 		const allResult = await this.findAll();
 		if (allResult.isErr()) {
 			return err(allResult.error);
@@ -68,9 +76,12 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * タスクを保存する（作成または更新）
 	 */
-	async save(task: Task): Promise<Result<Task, TaskNotFoundError>> {
-		const textResult = this.documentClient.getActiveDocumentText();
+	async save(task: Task): Promise<Result<Task, TaskNotFoundError | NoActiveEditorError>> {
+		const textResult = await this.documentClient.getCurrentDocumentText();
 		if (textResult.isErr()) {
+			if (textResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskNotFoundError(task.id));
 		}
 
@@ -108,6 +119,9 @@ export class MarkdownTaskRepository implements TaskRepository {
 
 		const writeResult = await this.documentClient.replaceDocumentText(editResult.value);
 		if (writeResult.isErr()) {
+			if (writeResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskNotFoundError(task.id));
 		}
 
@@ -117,9 +131,12 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * タスクを削除する
 	 */
-	async delete(id: string): Promise<Result<void, TaskNotFoundError>> {
-		const textResult = this.documentClient.getActiveDocumentText();
+	async delete(id: string): Promise<Result<void, TaskNotFoundError | NoActiveEditorError>> {
+		const textResult = await this.documentClient.getCurrentDocumentText();
 		if (textResult.isErr()) {
+			if (textResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskNotFoundError(id));
 		}
 
@@ -145,6 +162,9 @@ export class MarkdownTaskRepository implements TaskRepository {
 
 		const writeResult = await this.documentClient.replaceDocumentText(editResult.value);
 		if (writeResult.isErr()) {
+			if (writeResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskNotFoundError(id));
 		}
 
@@ -154,9 +174,12 @@ export class MarkdownTaskRepository implements TaskRepository {
 	/**
 	 * 利用可能なパス（見出し階層）を全て取得する
 	 */
-	async getAvailablePaths(): Promise<Result<Path[], TaskParseError>> {
-		const textResult = this.documentClient.getActiveDocumentText();
+	async getAvailablePaths(): Promise<Result<Path[], TaskParseError | NoActiveEditorError>> {
+		const textResult = await this.documentClient.getCurrentDocumentText();
 		if (textResult.isErr()) {
+			if (textResult.error instanceof InfraNoActiveEditorError) {
+				return err(new NoActiveEditorError());
+			}
 			return err(new TaskParseError(0, textResult.error.message));
 		}
 

--- a/packages/core/src/infrastructure/clients/index.ts
+++ b/packages/core/src/infrastructure/clients/index.ts
@@ -16,6 +16,7 @@ export {
 	DocumentEditError,
 	type DocumentInfo,
 	DocumentNotFoundError,
+	NoActiveEditorError,
 	VscodeDocumentClient,
 	type VscodeDocumentDeps,
 } from './vscodeDocumentClient';

--- a/packages/core/src/infrastructure/clients/vscodeDocumentClient.test.ts
+++ b/packages/core/src/infrastructure/clients/vscodeDocumentClient.test.ts
@@ -3,6 +3,7 @@ import type * as vscode from 'vscode';
 import {
 	DocumentEditError,
 	DocumentNotFoundError,
+	NoActiveEditorError,
 	VscodeDocumentClient,
 	type VscodeDocumentDeps,
 } from './vscodeDocumentClient';
@@ -46,7 +47,7 @@ describe('VscodeDocumentClient', () => {
 
 			expect(result.isErr()).toBe(true);
 			if (result.isErr()) {
-				expect(result.error).toBeInstanceOf(DocumentNotFoundError);
+				expect(result.error).toBeInstanceOf(NoActiveEditorError);
 				expect(result.error.message).toBe('アクティブなエディタがありません');
 			}
 		});
@@ -166,7 +167,7 @@ describe('VscodeDocumentClient', () => {
 
 			expect(result.isErr()).toBe(true);
 			if (result.isErr()) {
-				expect(result.error).toBeInstanceOf(DocumentNotFoundError);
+				expect(result.error).toBeInstanceOf(NoActiveEditorError);
 			}
 		});
 

--- a/packages/core/src/infrastructure/clients/vscodeDocumentClient.ts
+++ b/packages/core/src/infrastructure/clients/vscodeDocumentClient.ts
@@ -13,6 +13,18 @@ export class DocumentNotFoundError extends Error {
 }
 
 /**
+ * アクティブなエディタがないエラー
+ * これは正常な状態遷移の一つ（WebViewにフォーカスがある場合など）
+ */
+export class NoActiveEditorError extends Error {
+	readonly _tag = 'NoActiveEditorError';
+	constructor(message = 'アクティブなエディタがありません') {
+		super(message);
+		this.name = 'NoActiveEditorError';
+	}
+}
+
+/**
  * ドキュメント編集エラー
  */
 export class DocumentEditError extends Error {
@@ -55,15 +67,32 @@ export interface DocumentInfo {
  * VSCodeのテキストドキュメント操作のラッパー
  */
 export class VscodeDocumentClient {
+	private currentDocumentUri: vscode.Uri | undefined;
+
 	constructor(private readonly deps: VscodeDocumentDeps) {}
+
+	/**
+	 * 現在のドキュメントURIを設定する
+	 * WebViewパネル作成時やエディタ切り替え時に呼び出される
+	 */
+	setCurrentDocumentUri(uri: vscode.Uri | undefined): void {
+		this.currentDocumentUri = uri;
+	}
+
+	/**
+	 * 現在のドキュメントURIを取得する
+	 */
+	getCurrentDocumentUri(): vscode.Uri | undefined {
+		return this.currentDocumentUri;
+	}
 
 	/**
 	 * アクティブなドキュメントの情報を取得する
 	 */
-	getActiveDocument(): Result<DocumentInfo, DocumentNotFoundError> {
+	getActiveDocument(): Result<DocumentInfo, NoActiveEditorError> {
 		const editor = this.deps.getActiveTextEditor();
 		if (!editor) {
-			return err(new DocumentNotFoundError('アクティブなエディタがありません'));
+			return err(new NoActiveEditorError());
 		}
 
 		const document = editor.document;
@@ -78,7 +107,7 @@ export class VscodeDocumentClient {
 	/**
 	 * アクティブなドキュメントのテキストを取得する
 	 */
-	getActiveDocumentText(): Result<string, DocumentNotFoundError> {
+	getActiveDocumentText(): Result<string, NoActiveEditorError> {
 		const result = this.getActiveDocument();
 		return result.map((doc) => doc.text);
 	}
@@ -86,42 +115,79 @@ export class VscodeDocumentClient {
 	/**
 	 * アクティブなドキュメントのURIを取得する
 	 */
-	getActiveDocumentUri(): Result<vscode.Uri, DocumentNotFoundError> {
+	getActiveDocumentUri(): Result<vscode.Uri, NoActiveEditorError> {
 		const result = this.getActiveDocument();
 		return result.map((doc) => doc.uri);
 	}
 
 	/**
-	 * アクティブなドキュメントのテキストを置換する
+	 * 現在のドキュメントのテキストを取得する
+	 * currentDocumentUriがセットされている場合はそれを使用し、
+	 * なければactiveTextEditorから取得する
+	 */
+	async getCurrentDocumentText(): Promise<
+		Result<string, NoActiveEditorError | DocumentNotFoundError>
+	> {
+		// currentDocumentUriがセットされている場合はそれを使用
+		if (this.currentDocumentUri) {
+			return this.openAndGetText(this.currentDocumentUri);
+		}
+
+		// フォールバック: activeTextEditorから取得
+		return this.getActiveDocumentText();
+	}
+
+	/**
+	 * 現在のドキュメントのURIを取得する
+	 * currentDocumentUriがセットされている場合はそれを使用し、
+	 * なければactiveTextEditorから取得する
+	 */
+	getCurrentDocumentUriOrActive(): Result<vscode.Uri, NoActiveEditorError> {
+		if (this.currentDocumentUri) {
+			return ok(this.currentDocumentUri);
+		}
+		return this.getActiveDocumentUri();
+	}
+
+	/**
+	 * 現在のドキュメントのテキストを置換する
+	 * currentDocumentUriがセットされている場合はそれを使用し、
+	 * なければactiveTextEditorから取得する
 	 */
 	async replaceDocumentText(
 		newText: string,
-	): Promise<Result<void, DocumentNotFoundError | DocumentEditError>> {
-		const uriResult = this.getActiveDocumentUri();
+	): Promise<Result<void, NoActiveEditorError | DocumentNotFoundError | DocumentEditError>> {
+		// URIを取得
+		const uriResult = this.getCurrentDocumentUriOrActive();
 		if (uriResult.isErr()) {
 			return err(uriResult.error);
 		}
-
 		const uri = uriResult.value;
-		const docResult = this.getActiveDocument();
-		if (docResult.isErr()) {
-			return err(docResult.error);
+
+		// ドキュメントを開いてテキストと行数を取得
+		try {
+			const document = await this.deps.openTextDocument(uri);
+			const lineCount = document.lineCount;
+			const lastLine = lineCount > 0 ? lineCount - 1 : 0;
+			const lastLineText = document.getText().split('\n')[lastLine] ?? '';
+
+			const edit = this.deps.createWorkspaceEdit();
+			const fullRange = this.deps.createRange(0, 0, lastLine, lastLineText.length);
+			edit.replace(uri, fullRange, newText);
+
+			const success = await this.deps.applyEdit(edit);
+			if (!success) {
+				return err(new DocumentEditError('ドキュメントの編集に失敗しました'));
+			}
+
+			return ok(undefined);
+		} catch (error) {
+			return err(
+				new DocumentNotFoundError(
+					`ドキュメントを開けませんでした: ${error instanceof Error ? error.message : String(error)}`,
+				),
+			);
 		}
-
-		const lineCount = docResult.value.lineCount;
-		const lastLine = lineCount > 0 ? lineCount - 1 : 0;
-		const lastLineText = docResult.value.text.split('\n')[lastLine] ?? '';
-
-		const edit = this.deps.createWorkspaceEdit();
-		const fullRange = this.deps.createRange(0, 0, lastLine, lastLineText.length);
-		edit.replace(uri, fullRange, newText);
-
-		const success = await this.deps.applyEdit(edit);
-		if (!success) {
-			return err(new DocumentEditError('ドキュメントの編集に失敗しました'));
-		}
-
-		return ok(undefined);
 	}
 
 	/**

--- a/packages/core/src/interface/adapters/taskController.ts
+++ b/packages/core/src/interface/adapters/taskController.ts
@@ -10,6 +10,7 @@ import type {
 } from '../../application/usecases';
 import type { Task, TaskMetadata } from '../../domain/entities/task';
 import type { InvalidStatusError } from '../../domain/errors/invalidStatusError';
+import type { NoActiveEditorError } from '../../domain/errors/noActiveEditorError';
 import type { TaskNotFoundError } from '../../domain/errors/taskNotFoundError';
 import type { TaskParseError } from '../../domain/errors/taskParseError';
 import { Path } from '../../domain/valueObjects/path';
@@ -52,7 +53,7 @@ export class TaskController {
 	/**
 	 * 全タスクを取得する
 	 */
-	async getTasks(): Promise<Result<TaskDto[], TaskParseError>> {
+	async getTasks(): Promise<Result<TaskDto[], TaskParseError | NoActiveEditorError>> {
 		const result = await this.getTasksUseCase.execute();
 		return result.map((tasks) => tasks.map(this.toDto));
 	}
@@ -62,7 +63,7 @@ export class TaskController {
 	 */
 	async createTask(
 		dto: CreateTaskDto,
-	): Promise<Result<TaskDto, TaskNotFoundError | InvalidStatusError>> {
+	): Promise<Result<TaskDto, TaskNotFoundError | InvalidStatusError | NoActiveEditorError>> {
 		// ステータスの変換
 		let status: Status | undefined;
 		if (dto.status !== undefined) {
@@ -89,7 +90,7 @@ export class TaskController {
 	 */
 	async updateTask(
 		dto: UpdateTaskDto,
-	): Promise<Result<TaskDto, TaskNotFoundError | TaskParseError>> {
+	): Promise<Result<TaskDto, TaskNotFoundError | TaskParseError | NoActiveEditorError>> {
 		const input: UpdateTaskInput = {
 			id: dto.id,
 			title: dto.title,
@@ -104,7 +105,7 @@ export class TaskController {
 	/**
 	 * タスクを削除する
 	 */
-	async deleteTask(id: string): Promise<Result<void, TaskNotFoundError>> {
+	async deleteTask(id: string): Promise<Result<void, TaskNotFoundError | NoActiveEditorError>> {
 		return this.deleteTaskUseCase.execute(id);
 	}
 
@@ -114,7 +115,9 @@ export class TaskController {
 	async changeTaskStatus(
 		id: string,
 		newStatus: string,
-	): Promise<Result<TaskDto, TaskNotFoundError | TaskParseError | InvalidStatusError>> {
+	): Promise<
+		Result<TaskDto, TaskNotFoundError | TaskParseError | InvalidStatusError | NoActiveEditorError>
+	> {
 		const statusResult = Status.create(newStatus);
 		if (statusResult.isErr()) {
 			return err(statusResult.error);

--- a/packages/core/src/interface/adapters/webViewMessageHandler.ts
+++ b/packages/core/src/interface/adapters/webViewMessageHandler.ts
@@ -142,7 +142,12 @@ export class WebViewMessageHandler {
 	 * エラーを送信する
 	 */
 	private sendError(error: Error & { _tag?: string }): void {
-		logger.error(`Error: ${error.message}`, { errorType: error._tag, message: error.message });
+		// NoActiveEditorErrorは正常な状態遷移の一つなのでdebugレベルでログ出力
+		if (error._tag === 'NoActiveEditorError') {
+			logger.debug(`No active editor: ${error.message}`);
+		} else {
+			logger.error(`Error: ${error.message}`, { errorType: error._tag, message: error.message });
+		}
 		this.messageClient.sendError(error.message, error._tag);
 	}
 }

--- a/packages/webview/src/hooks/useVscodeApi.ts
+++ b/packages/webview/src/hooks/useVscodeApi.ts
@@ -190,12 +190,21 @@ export function useKanban() {
 					tasks: prev.tasks.filter((t) => t.id !== payload.id),
 				}));
 			},
-			ERROR: (payload: { message: string }) => {
-				setState((prev) => ({
-					...prev,
-					error: payload.message,
-					isLoading: false,
-				}));
+			ERROR: (payload: { message: string; code?: string }) => {
+				// NoActiveEditorErrorは正常な状態遷移なのでエラーとして扱わない
+				if (payload.code === 'NoActiveEditorError') {
+					setState((prev) => ({
+						...prev,
+						error: 'Markdownファイルを開いてください',
+						isLoading: false,
+					}));
+				} else {
+					setState((prev) => ({
+						...prev,
+						error: payload.message,
+						isLoading: false,
+					}));
+				}
 			},
 		}),
 		[],

